docs: add example to exchange digital twins (#313)
eclipse-tractusx/tractus-x-umbrella: docs/user/guides/digitaltwin-exchange/provide-digitaltwins.md

# Provide Digital Twins

This documentation guides you through the process of creating and providing digital twins in a local dataspace using the Eclipse Dataspace Connector (EDC).
You need to configure the following components to implement this example: 
    
* **Digital Twin Registry (DTR)** - to create and publish digital twins
* **Submodel Server** - to create and provide data (so-called submodels) for digital twins based on aspect models
* **EDC** - to connect DTR and Submodel Server to your local dataspace. Both components, DTR and Submodel Server are only accessible via the EDC.

> ![Setup_Example](images/Setup_Example.drawio.svg)

Reviewer: Would it be possible (if not much work) to modify the background color of Bobs and Alice actions, hard to read white text over it.

The following command can be utilized to create the access policy. 
```bash
curl -L -X POST 'http://dataprovider-controlplane.tx.test/management/v3/policydefinitions' \
  -H 'Content-Type: application/json' \
  -H 'X-Api-Key: TEST2' \
  --data-raw '{
    "@context": {
      "odrl": "http://www.w3.org/ns/odrl/2/"
    },
    "@type": "PolicyDefinitionRequestDto",
    "@id": "access-policy",
    "policy": {
      "@type": "odrl:Set",
      "odrl:permission": [
        {
          "odrl:action": "USE",
          "odrl:constraint": {
            "@type": "LogicalConstraint",
            "odrl:or": [
              {
                "@type": "Constraint",
                "odrl:leftOperand": {
                  "@id": "BusinessPartnerNumber"
                },
                "odrl:operator": {
                  "@id": "odrl:eq"
                },
                "odrl:rightOperand": "BPNL00000003AZQP"
              }
            ]
          }
        }
      ]
    }
  }' | jq
  ```
### Usage Policy for dDTR

> The usage policy defines the conditions under which data from the decentralized Digital Twin Registry (dDTR) may be utilized. By using a standardized purpose such as **"cx.core.digitalTwinRegistry:1"**, it ensures that all providers within the Catena-X ecosystem employ a consistent usage policy. This enhances the transparency and trustworthiness of the system.

Reviewer: ```suggestion
The usage policy defines the conditions under which data from the decentralized Digital Twin Registry (dDTR) may be utilized. By using a standardized purpose, such as **"cx.core.digitalTwinRegistry:1"**, it ensures that all providers within the Catena-X ecosystem employ a consistent usage policy. This enhances the transparency and trustworthiness of the system.
```

```bash
curl -L -X POST 'http://dataprovider-controlplane.tx.test/management/v3/policydefinitions' \
  -H 'Content-Type: application/json' \
  -H 'X-Api-Key: TEST2' \
  --data-raw '{
  "@context": {
    "@vocab": "https://w3id.org/edc/v0.0.1/ns/",
    "edc": "https://w3id.org/edc/v0.0.1/ns/",
    "cx-policy": "https://w3id.org/catenax/policy/",
    "odrl": "http://www.w3.org/ns/odrl/2/"
  },
  "@id": "USEAGEPOLICYIDfordDTR",
  "policy": {
    "@type": "odrl:Set",
    "odrl:permission": {
      "odrl:action": {
        "@id": "odrl:use"
      },
      "odrl:constraint": {
        "odrl:and": [          
          {
            "odrl:leftOperand": {
              "@id": "cx-policy:UsagePurpose"
            },
            "odrl:operator": {
              "@id": "odrl:eq"
            },
            "odrl:rightOperand": "cx.core.digitalTwinRegistry:1"
          }
        ]
      }
    }
  }
}' | jq
```
## Creating a dDTR Contract Definition

> In common a contract definition always consists out of 1 to n assets, an access policy and 1 to n usage policies. So you should first create a access and usage policy for your registry (like shown above). In common policies can be reused, but please do not reuse the usage policy of the registry for other contract definitions as the purpose of the usage is focused on the dDTR use!

Reviewer: ```suggestion
In common, a contract definition always consists out of 1 to n assets, an access policy and 1 to n usage policies. So you should first create an access and usage policy for your registry (like shown above). In common policies can be reused, but please do not reuse the usage policy of the registry for other contract definitions as the purpose of the usage is focused on the dDTR use!
```

```bash
curl -L -X POST 'http://dataprovider-controlplane.tx.test/management/v3/contractdefinitions' \
  -H 'Content-Type: application/json' \
  -H 'X-Api-Key: TEST2' \
  --data-raw '{
  "@id": "Contract-Definition-Digital-Twin-Registry",
  "@type": "ContractDefinition",
  "accessPolicyId": "access-policy",
  "contractPolicyId": "USEAGEPOLICYIDfordDTR",
  "assetsSelector": {
    "@type": "Criterion",
    "operandLeft": "http://purl.org/dc/terms/type",
    "operator": "like",
    "operandRight": "%https://w3id.org/catenax/taxonomy#DigitalTwinRegistry%"
  },
  "@context": {
    "@vocab": "https://w3id.org/edc/v0.0.1/ns/",
    "edc": "https://w3id.org/edc/v0.0.1/ns/",
    "tx": "https://w3id.org/tractusx/v0.0.1/ns/",
    "tx-auth": "https://w3id.org/tractusx/auth/",
    "cx-policy": "https://w3id.org/catenax/policy/",
    "odrl": "http://www.w3.org/ns/odrl/2/"
  }
}' | jq
```
## Filter for the dDTR Offer

> If everything is setup like described above, a data consumer should find the dDTR offer in the catalogue with the the catalogue request with the filter expression shown in the example. The filter uses the dct type to filter for the registry offer.

Reviewer: ```suggestion
If everything is set up like described above, a data consumer should find the dDTR offer in the catalogue with the catalogue request with the filter expression shown in the example. The filter uses the dct type to filter for the registry offer.
```
